File: packages/typescript/terraform/src/stack-group.ts
import {
  BaseConfig,
  StackGroup as BaseStackGroup,
  StackGroupLoader,
} from '@awslv/cdk-organizer-core';
import { App } from 'cdktf';

export class StackGroup<
  ConfigType extends BaseConfig = BaseConfig
> extends BaseStackGroup<App, ConfigType> {
  constructor(app: App, loader: StackGroupLoader, module: string) {
    super(app, loader, module);
  }
}

File: packages/python/cdk_organizer/terraform/stack_group.py
"""Terraform CDK Stack Group Class."""

from typing import TypeVar

from cdk_organizer.stack_group import StackGroup as BaseStackGroup
from cdk_organizer.stack_group import StackGroupLoader
from cdktf import App

CDK_CONFIG_TYPE = TypeVar('CDK_CONFIG_TYPE', covariant=True)


class StackGroup(BaseStackGroup[App, CDK_CONFIG_TYPE]):
    """Stack Group for Terraform CDK."""

    def __init__(self, app: App, loader: StackGroupLoader) -> None:
        """Stack group constructor."""
        super().__init__(app, loader)
